src/InnerResultIterator.php

```diff
@@ -241,6 +241,9 @@ public function next(): void
                 $dbRow = $this->objectStorage->get($mainBeanTableName, $hash);
                 if ($dbRow !== null) {
                     $bean = $dbRow->getTDBMObject();
+                    if ($bean->_getStatus() === TDBMObjectStateEnum::STATE_NOT_LOADED) {
```

Comment:
I used `STATE_NOT_LOADEDD` here, but there are other states maybe some are relevant but I feel like this way is the less risky.